system/Model.php

```diff
@@ -40,47 +41,60 @@
+ * @method $this                              offset(int $offset)
+ * @method $this                              onConstraint(array|RawSql|string $set = null)
+ * @method $this                              orderBy(string $orderBy, string $direction = '', ?bool $escape = null)
+ * @method $this                              orGroupStart()
+ * @method $this                              orHaving($key, $value = null, ?bool $escape = null)
+ * @method $this                              orHavingGroupStart()
+ * @method $this                              orHavingIn(?string $key = null, $values = null, ?bool $escape = null)
+ * @method $this                              orHavingLike($field, string $match = '', string $side = 'both', ?bool $escape = null, bool $insensitiveSearch = false)
+ * @method $this                              orHavingNotIn(?string $key = null, $values = null, ?bool $escape = null)
+ * @method $this                              orLike($field, string $match = '', string $side = 'both', ?bool $escape = null, bool $insensitiveSearch = false)
+ * @method $this                              orNotGroupStart()
+ * @method $this                              orNotHavingGroupStart()
+ * @method $this                              orNotHavingLike($field, string $match = '', string $side = 'both', ?bool $escape = null, bool $insensitiveSearch = false)
+ * @method $this                              orNotLike($field, string $match = '', string $side = 'both', ?bool $escape = null, bool $insensitiveSearch = false)
+ * @method $this                              orWhere($key, $value = null, ?bool $escape = null)
+ * @method $this                              orWhereIn(?string $key = null, $values = null, ?bool $escape = null)
+ * @method $this                              orWhereNotIn(?string $key = null, $values = null, ?bool $escape = null)
+ * @method $this                              select($select = '*', ?bool $escape = null)
+ * @method $this                              selectAvg(string $select = '', string $alias = '')
+ * @method $this                              selectCount(string $select = '', string $alias = '')
+ * @method $this                              selectMax(string $select = '', string $alias = '')
+ * @method $this                              selectMin(string $select = '', string $alias = '')
+ * @method $this                              selectSum(string $select = '', string $alias = '')
+ * @method $this                              setQueryAsData(BaseBuilder|RawSql $query, ?string $alias = null, array|string|null $columns = null)
+ * @method bool                               truncate()
+ * @method $this                              union(BaseBuilder|Closure $union)
+ * @method $this                              unionAll(BaseBuilder|Closure $union)
+ * @method BaseResult|false|int|object|string updateBatch(array|object|null $set = null, array|RawSql|string|null $constraints = null, int $batchSize = 100)
```

Comment:
This returns `BaseResult|false|int|object|string`. So the model does not work correctly.